LibreHardwareMonitorLib/Hardware/Cpu/CpuLoad.cs

```diff
@@ -42,32 +44,92 @@ private static bool GetTimes(out long[] idle, out long[] total)
+            {
+                return false;
+            }
+
+            //Query processor performance information
+            Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION[] perfInformation = new Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION[64];
+            int perfSize = Marshal.SizeOf(typeof(Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION));
+            if (Interop.NtDll.NtQuerySystemInformation(Interop.NtDll.SYSTEM_INFORMATION_CLASS.SystemProcessorPerformanceInformation, perfInformation, perfInformation.Length * perfSize, out int perfReturn) != 0)
+            {
+                return false;
+            }
+
+            idle = new long[idleReturn / idleSize];
+            for (int i = 0; i < idle.Length; i++)
+            {
+                idle[i] = idleInformation[i].IdleTime;
+            }
+
+            total = new long[perfReturn / perfSize];
+            for (int i = 0; i < total.Length; i++)
+            {
+                total[i] = perfInformation[i].KernelTime + perfInformation[i].UserTime;
+            }
         }
-
-        //Query processor performance information
-        Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION[] perfInformation = new Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION[64];
-        int perfSize = Marshal.SizeOf(typeof(Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION));
-        if (Interop.NtDll.NtQuerySystemInformation(Interop.NtDll.SYSTEM_INFORMATION_CLASS.SystemProcessorPerformanceInformation, perfInformation, perfInformation.Length * perfSize, out int perfReturn) != 0)
-        {
-            return false;
-        }
-
-        idle = new long[idleReturn / idleSize];
-        for (int i = 0; i < idle.Length; i++)
-        {
-            idle[i] = idleInformation[i].IdleTime;
-        }
-
-        total = new long[perfReturn / perfSize];
-        for (int i = 0; i < total.Length; i++)
+        if (Software.OperatingSystem.IsUnix)
         {
-            total[i] = perfInformation[i].KernelTime + perfInformation[i].UserTime;
+            List<long> idleList = new();
+            List<long> totalList = new();
+            // Get the information
+            string[] cpuInfo = File.ReadAllLines("/proc/stat");
+            // currently parse the OverAll CPU info
+            // cpu   1583083 737    452845   36226266 723316   63685 31896     0       0       0
+            // cpu0  397468  189    109728   9040007  191429   16939 14954     0       0       0
+            // 0=cpu 1=user  2=nice 3=system 4=idle   5=iowait 6=irq 7=softirq 8=steal 9=guest 10=guest_nice
+            foreach (string cpuinfo in cpuInfo.Where(s => s.StartsWith("cpu")).Where(s => s[3] != ' '))
+            {
+                string[] overall = cpuinfo.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I think this should also check if there are enough values available (11).

Reply:
the try/catch clause in the below will handle if there are not enough values available.

Reply:
restuctured